include/nbl/builtin/hlsl/workgroup/shuffle.hlsl

```diff
@@ -0,0 +1,40 @@
+#ifndef _NBL_BUILTIN_HLSL_WORKGROUP_SHUFFLE_INCLUDED_
+#define _NBL_BUILTIN_HLSL_WORKGROUP_SHUFFLE_INCLUDED_
+
+#include "nbl/builtin/hlsl/memory_accessor.hlsl"
+
+// TODO: Add other shuffles
+// TODO: Consider adding an enable_if or static assert that 1 <= N <= 4 and that Scalar is a proper scalar type
+// TODO: Consider adding version that doesn't take a precomputed threadID and instead calls workgroup::SubgroupContiguousIndex
+
+// Unlike subgroups we pass a precomputed threadID so we don't go around recomputing it every time
```

Comment:
make it consistent with other workgroup ops, and a struct

common subexpression elimination (CSE) during spirv-opt should take care of duplicate `threadID` computation, provide a variant that does not ask for `threadID` to be consistent with other workgroup ops (ballot, arithmetic, etc)

IMO you could implement it in terms of a shuffle